src/hotspot/share/cds/filemap.cpp

```diff
@@ -2047,6 +2047,14 @@ void FileMapInfo::fixup_mapped_heap_regions() {
     assert(open_archive_heap_ranges != NULL, "NULL open_archive_heap_ranges array with non-zero count");
     G1CollectedHeap::heap()->fill_archive_regions(open_archive_heap_ranges,
                                                   num_open_archive_heap_ranges);
+
+    // Populate the open archive regions' G1BlockOffsetTableParts. That ensures
+    // fast G1BlockOffsetTablePart::block_start operations for a given address
+    // within the open archive regions during G1RemSet::refine_card_concurrently.
```

Comment:
Not only refinement is subject to the issue but any operation trying to find start of an object (e.g. card table scanning). Please fix the comment accordingly.

Reply:
Updated comment based on the feedback from Thomas. Thanks for the review.